### PR TITLE
Arc decoder

### PR DESCRIPTION
**Pull request by quantumjim**
### Summary

Adds `decoders.hdrg_decoders.ClusteringDecoder`. This can be applied to various codes in general, as long as they have a `check_nodes` method which assesses the neutraility of a given set of nodes. Currently it is compatible with `RepetitionCodeCircuit` and `ArcCodeCircuit`, and these have been made compatible with it.

It is currently the only decoder in Qiskit-QEC that works with `ArcCodeCircuit`.

### Details and comments

* Based on the decoder presented in [this lecture](https://github.com/quantumjim/qec_lectures/blob/main/lecture-4.ipynb).
* I can promote the idea of people contributing to this decoder in early December (assuming a merge).

### CODE REVIEW

**Comment by quantumjim** [comment]
Weirdness undone!

File: qiskit_qec/decoders/hdrg_decoders.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
```
Comment (grace-harper-ibm) [inline qiskit_qec/decoders/hdrg_decoders.py:1]: If there are going to be multiple hdrg decoders, do we want to just make a subfolder
`decoders/hdrg_decoders/clustering_decoder.py`  so each decoder still gets their own file?

**Review by grace-harper-ibm — APPROVED** [review]
Let me know your thoughts on the subfolder.

**Review by grace-harper-ibm — APPROVED** [review]
LGTM

**Comment by quantumjim** [comment]
I don't see there being enough HDRG decoders for a subfolder. I think they could just share a file, if that sounds reasonable.

**Pull request merged**